indexing-hadoop/src/main/java/io/druid/indexer/HadoopDruidIndexerConfig.java

```diff
@@ -316,6 +316,14 @@ public HadoopyShardSpec getShardSpec(Bucket bucket)
     return schema.getTuningConfig().getShardSpecs().get(bucket.time).get(bucket.partitionNum);
   }
 
+  /**
+   * Job instance should have Configuration set (by calling {@link #addJobProperties(Job)}
```

Comment:
can you think of a way we can programmatically check for that, so that we can throw an ISE in case the configuration is not set before calling addInputPaths?